### PR TITLE
msdfgen: conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/msdfgen/all/conanfile.py [hunk 1]
```diff
@@ -29,85 +34,94 @@ class MsdfgenConan(ConanFile):
...
+        deps.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        apply_conandata_patches(self)
+        cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
+        # workaround against CMAKE_FIND_PACKAGE_PREFER_CONFIG ON in conan toolchain
+        replace_in_file(self, cmakelists, "find_package(Freetype REQUIRED)", "find_package(Freetype REQUIRED MODULE)")
```
Comment (prince-chrismc) [inline recipes/msdfgen/all/conanfile.py:86]: Can you pretty please move some of these to patches?

I'd really love to have the patch type so we can track the changes we need to make 🙏

**Review by prince-chrismc — COMMENTED** [review]
As always your work is amazing, I;d just love to see "fix find deps for conan" and "unvendor deps" as patches 🙏

File: recipes/msdfgen/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/msdfgen/all/conanfile.py:86]: it's a patch we have to keep forever, so it's not a good idea to move it to a patch file.

Comment (prince-chrismc) [reply recipes/msdfgen/all/conanfile.py:86]: They are difficult to review, we can not easily see what effect its having and what changes are being made. They are also difficult to track and understand for consumers.

We are starting to have the conversation about security in CCI so we may start adding policies in place to improve that.


For example removing inactive users from the access list. Is a recent thing.


I understand you point about maintaining patches but security is more important, I think most people would agree

**Comment by SpaceIm** [comment]
It would be nice to not be closed by bot for details unrelated to PR itself (there was already replace_in_file in this recipe).

**Review by prince-chrismc — APPROVED** [review]
This looks amazing

**Review by uilianries — APPROVED** [review]

**Pull request merged**